src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -248,38 +291,59 @@ public ResponseEntity getLectureAttachment(@PathVariable Long lectureId, @PathVa
         if (!optionalLecture.isPresent()) {
             return ResponseEntity.badRequest().build();
         }
+        if (!validateTemporaryAccessToken(temporaryAccessToken, filename)) {
+            String errorMessage = "You don't have the access rights for this file! Please login to Artemis and download the attachment in the corresponding lecture";
+            return ResponseEntity.status(403).body(errorMessage);
+        }
+        return buildFileResponse(Constants.LECTURE_ATTACHMENT_FILEPATH + optionalLecture.get().getId(), filename);
+    }
+
+    /**
+     * Validates temporary access token
+     * @param temporaryAccessToken token to be validated
+     * @param filename the name of the file
+     * @return true if temporaryAccessToken is valid for this file, false otherwise
+     */
+    private boolean validateTemporaryAccessToken(String temporaryAccessToken, String filename) {
         if (temporaryAccessToken == null || !this.tokenProvider.validateTokenForAuthorityAndFile(temporaryAccessToken, TokenProvider.DOWNLOAD_FILE_AUTHORITY, filename)) {
             log.info("Attachment with invalid token was accessed");
-            return ResponseEntity.status(403)
-                    .body("You don't have the access rights for this file! Please login to Artemis and download the attachment in the corresponding lecture");
+            return false;
         }
-        Lecture lecture = optionalLecture.get();
+        return true;
+    }
+
+    /**
+     * Builds the response with headers, body and content type for specified path and file name
+     * @param path to the file
+     * @param filename the name of the file
+     * @return response entity
+     */
+    private ResponseEntity buildFileResponse(String path, String filename) {
         try {
-            byte[] file = fileService.getFileForPath(Constants.LECTURE_ATTACHMENT_FILEPATH + lecture.getId() + '/' + filename);
+            byte[] file = fileService.getFileForPath(path + '/' + filename);
             if (file == null) {
                 return ResponseEntity.notFound().build();
             }
-            ByteArrayResource resource = new ByteArrayResource(file);
 
+            ByteArrayResource resource = new ByteArrayResource(file);
             ContentDisposition contentDisposition = ContentDisposition.builder("inline").filename(filename).build();
-
             HttpHeaders headers = new HttpHeaders();
             headers.setContentDisposition(contentDisposition);
-            String mediaType = "application/pdf";
-            if (filename.endsWith(".pdf")) {
-                mediaType = "application/pdf";
+            FileNameMap fileNameMap = URLConnection.getFileNameMap();
+            String mimeType = fileNameMap.getContentTypeFor(filename);
```

Comment:
Did you test that this works for pdf and zip files for lectures as well?

Reply:
Yes this works